synapse/api/filtering.py

```diff
@@ -181,6 +184,15 @@ class Filter(object):
     def __init__(self, filter_json):
         self.filter_json = filter_json
 
+    def list_rooms(self):
+        """The list of room_id strings this filter restricts the output to
+        or None if the this filter doesn't list the room ids.
+        """
+        if "rooms" in self.filter_json:
+            return list(set(self.filter_json["rooms"]))
+        else:
+            return None
```

Comment:
We check that the list is non-None before calling into these handlers. I'd rather have a hard error here than a default.